types/lodash/common/lang.d.ts

```diff
@@ -1098,6 +1098,10 @@ declare module "../index" {
          * // => false
          */
         isObjectLike(value?: any): boolean;
+
+        // NOTE: The seemingly logical `value is object` type predicate cannot
+        // be added, as it would break the negative version of the type guard.
+        // https://github.com/DefinitelyTyped/DefinitelyTyped/pull/69915#pullrequestreview-2143148704
```

Comment:
I explained in a little more detail here: https://github.com/DefinitelyTyped/DefinitelyTyped/discussions/68867#discussioncomment-8671502. In fact it looks like my past self corrects me... The type guard should work fine on isObjectLike, just not isPlainObject. Sorry for the confusion!

Reply:
Sorry for the long delay, I have finally got to wrapping this up. In the latest commit I add the type guard to `isObjectLike` and the explanation to `isPlainObject`.